### ISSUE TITLE
Linux Steam binary is missing libGLEW library

### ISSUE THREAD

**Issue by RadhiFadlillah** [endless-sky/endless-sky#5135]
**Describe the bug**

In my Manjaro Linux PC, Endless Sky that installed from Steam won't started after I run it. However, when I install it from my package manager, it runs properly without any issue. 

When I run the Steam binary from terminal, it gives me following output :

```
$ ./endless-sky

./endless-sky: error while loading shared libraries: libGLEW.so.1.10: cannot open shared object file: No such file or directory
```

True enough, after checking it using `ldd`, it's only missing that library :

```
$ ldd ./endless-sky

linux-vdso.so.1 (0x00007ffcff75a000)
libSDL2-2.0.so.0 => /usr/lib/libSDL2-2.0.so.0 (0x00007fc89de0f000)
libpng12.so.0 => /usr/lib/libpng12.so.0 (0x00007fc89dde3000)
libjpeg.so.8 => /usr/lib/libjpeg.so.8 (0x00007fc89dd4c000)
libGL.so.1 => /usr/lib/libGL.so.1 (0x00007fc89dcc6000)
libGLEW.so.1.10 => not found
libopenal.so.1 => /usr/lib/libopenal.so.1 (0x00007fc89db84000)
libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007fc89db60000)
libm.so.6 => /usr/lib/libm.so.6 (0x00007fc89da1b000)
libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007fc89da01000)
libc.so.6 => /usr/lib/libc.so.6 (0x00007fc89d83a000)
/lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007fc89dfb4000)
libdl.so.2 => /usr/lib/libdl.so.2 (0x00007fc89d834000)
libz.so.1 => /usr/lib/libz.so.1 (0x00007fc89d81a000)
libGLdispatch.so.0 => /usr/lib/libGLdispatch.so.0 (0x00007fc89d761000)
libGLX.so.0 => /usr/lib/libGLX.so.0 (0x00007fc89d72e000)
libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007fc89d551000)
libX11.so.6 => /usr/lib/libX11.so.6 (0x00007fc89d410000)
libxcb.so.1 => /usr/lib/libxcb.so.1 (0x00007fc89d3e6000)
libXau.so.6 => /usr/lib/libXau.so.6 (0x00007fc89d3df000)
libXdmcp.so.6 => /usr/lib/libXdmcp.so.6 (0x00007fc89d3d7000)
```

**Expected behavior**

Endless Sky runs properly.

**System (please complete the following information):**

 - OS: Manjaro Linux
 - Game Source: Steam
 - Version: 0.9.12

**Comment by 7eggert** [comment]
Same in Opensuse Tumbleweed; after installing the library from Debian Jessie, I get a SIGABORT.

**Comment by Eylok** [comment]
I was having the same issue and fixed it by installing "the glew" and "lib32-glew" packages.
However, after i installed those two i encountered another problem with the openal library, i reinstalled it and nothing. I ended up downloading the appimage from the releases page, extracted the appimage, and forced the game to use the libraries that come with it, and now it works fine. I don't know if it's an ideal solution to be honest, but it works.

**Comment by JohannPopper** [comment]
^ That's a severely overcomplicated way of running Endless Sky. Just use the AppImage. That's what it's for.

This isn't an Endless Sky bug. You are misunderstanding how to run apps on Linux. Software works by running on top of system libraries on which they depend. You are using a non-official way (Steam) to shoehorn an app built for one configuration into your random systems, so what you have is a complete app-library incompatibility. That's illogical. Steam only supports -- and thus ensures compatibility with -- the libraries of Ubuntu LTSes. Running Steam on Manjaro via Manjaro's rolling repo is inherently going to be unstable. This is what Flatpak and Snap were invented for. 

ES is distributed via official Ubuntu/Debian/Manjaro, etc, repos and PPA for Ubuntu, which are version tied to those specific system libraries per release. Use those, or cross-distro solutions like AppImage and flatpak. Installing within Steam is choosing to install a version of Endless Sky that is not tied to your OS or OS version system libraries, which is bound to fail because how can a game built for specific systems work on any random state of a rolling distro like Archjaro? It's also completely unnecessary, since ES is also built for cross-distro, cross-version use via AppImage or flatpak just for your use cases. 

So, to sum up, you can install ES through, 

1) your distro package manager
2) AppImage 
3) Flatpak.

You can ONLY install though Steam coherently if you have installed Steam via #1 on a supported OS or possibly #3 (if the Steam flatpak has been fixed), otherwise the underlying system libraries probably won't match the versions your version of ES was built for. I would say the most obvious simple solution is #2, and the ES devs agree with me, as it's one of the official direct ways they distribute the game in Github.

Like Admiral Kirk always says, Cadet, you have to learn how things work on a starship.

**Comment by MCOfficer** [comment]
I was under the impression that steam comes with its own runtime and thus its own libraries.

**Comment by JohannPopper** [comment]
Repo Steam includes some just like GOG installers, but there can still be versioning conflicts if the user's kernel conflicts, or if another systen dependency conflicts. In all likelihood, Steam just doesn't include glew. Now, the Flatpak Steam should include most required libraries on a stable base, but if it doesn't have glew built in, the game will try to grab the incompatible system version and fail. Thus, it's better to directly run ES via either its own Flatpak, AppImage, or even Snap if there is one. Steam implementation will always be shaky on non-supported distros for this reason, and AFAIK the Steam flatpak has had issues in the past because the publisher forgot a library or several, but that may be long resolved idk. There's probably a lot of interest in publishing a comprehensive and reliable the Steam flatpak for future game distribution that way, so it might already be long resolved (as well as proton libraries, vulkan, etc), and if Steam is your favored platform that's probably your best bet going forward. Good luck!

**Comment by gunqqer** [comment]
You can always try forcing the steam runtime, similar to forcing a specific version of proton, which make let you run the game without installing flatpak or something else

**Comment by tehhowch** [comment]
> When I run the steam binary from terminal

That's the problem. You are using your system dependencies and not the dependencies of the Steam runtime. If you wish to use the Steam-provided executable, you must tell it where it can find Steam's runtime:

```sh
Endless Sky $ ./endless-sky -v
./endless-sky: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory
Endless Sky $ ../../../ubuntu12_32/steam-runtime/run.sh ./endless-sky -v

Endless Sky ver. 0.9.13
License GPLv3+: GNU GPL version 3 or later: <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Compiled against SDL v2.0.14
Using SDL v2.0.14

Endless Sky $ pwd
~/.local/share/Steam/steamapps/common/Endless Sky
```

As such, this isn't a bug with the game or how it is shipped in Steam.

**Comment by 7eggert** [comment]
The library is NOT SHIPPED WITH STEAM.

**Comment by tehhowch** [comment]
It's listed in the runtime manifest.

https://repo.steampowered.com/steamrt-images-scout/snapshots/latest-container-runtime-depot/steam-runtime.manifest.txt

Are you using an unofficial wrapper? If so, please file an issue with that tool.

**Comment by 7eggert** [comment]
I just updated the seemingly official steam client, "Steam client application", built Jun 8 2021, Version 1623193086. However, I originally installed steam using the distribution's package (OpenSuse Tumbleweed), then it claimed to download all the current libraries (obviously it doesn't?).

https://bugzilla.opensuse.org/show_bug.cgi?id=1087706
